fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java

```diff
@@ -577,25 +579,30 @@ public void checkTimeout(long now) {
                 killConnection = true;
             }
         } else {
+            long timeout;
             if (userQueryTimeout > 0) {
                 // user set query_timeout property
-                if (delta > userQueryTimeout * 1000) {
-                    LOG.warn("kill query timeout, remote: {}, query timeout: {}",
-                            getMysqlChannel().getRemoteHostPortString(), userQueryTimeout);
-
-                    killFlag = true;
-                }
+                timeout = userQueryTimeout * 1000;
             } else {
                 // default use session query_timeout
-                if (delta > sessionVariable.getQueryTimeoutS() * 1000) {
-                    LOG.warn("kill query timeout, remote: {}, query timeout: {}",
-                            getMysqlChannel().getRemoteHostPortString(), sessionVariable.getQueryTimeoutS());
+                timeout = sessionVariable.getQueryTimeoutS() * 1000L;
+            }
 
-                    // Only kill
-                    killFlag = true;
+            //deal with insert stmt particularly
+            if (executor != null) {
+                StatementBase parsedStmt = executor.getParsedStmt();
+                if (parsedStmt instanceof InsertStmt) {
+                    timeout = sessionVariable.getInsertTimeoutS() * 1000L;
                 }
             }
+
+            if (delta > timeout) {
+                LOG.warn("kill query timeout, remote: {}, query timeout: {}",
```

Comment:
For INSERT statment, warn log should be about "insert timeout". Otherwise, user may increase query timeout after see this log when inserting and will be confused why it won't work.

Reply:
I was negligent, INSERT does need a different log msg. I'll modify it.